ports/qemu-riscv/Makefile

```diff
@@ -0,0 +1,140 @@
+include ../../py/mkenv.mk
+-include mpconfigport.mk
+
+# qstr definitions (must come before including py.mk)
+QSTR_DEFS = qstrdefsport.h
+
+# MicroPython feature configurations
+MICROPY_ROM_TEXT_COMPRESSION ?= 1
+
+# include py core make definitions
+include $(TOP)/py/py.mk
+include $(TOP)/extmod/extmod.mk
+
+BOARD ?= virt
+
+CROSS_COMPILE ?= riscv64-unknown-elf-
+
+# The GCC version string starts like this:
+# <target>-gcc (<optional git hash>) <major>.<minor>.<patch>
+GCC_VERSION = $(subst ., , $(word 3, $(shell $(CC) --version)))
+GCC_MAJOR = $(word 1, $(GCC_VERSION))
+GCC_MINOR = $(word 2, $(GCC_VERSION))
+GCC_PATCH = $(word 3, $(GCC_VERSION))
+
+ifeq ($(BOARD),virt)
+ABI=ilp32
+# GCC 10 and lower do not recognise the Zicsr extension in the
+# architecture name.  "Make" unfortunately does not provide any simple way
+# to perform numeric comparisons, so to keep things simple we assume that
+# GCC is at least version 10 for the time being.
+ifeq ($(GCC_MAJOR),10)
```

Comment:
Given this a brand new port on a pretty new arch, what do you think about dropping support for GCC 10 and older? It seems like a reasonable enough baseline requirement to me.

(If there's a commonly installed riscv toolchain out there that's older then we should keep it, although we'll also need to find a way to test for it in CI.)

Reply:
RISC-V support was officially added to GCC in version 7.5.0 - unofficial toolchains go as far back as 4.5 or so.  I'm not sure whether said toolchains can build this though :)

Anyway, this may require a bit more thought.  On one hand, the CI image (which I understand it being Ubuntu 22.04 LTS) ships GCC version 9, 10, 11, and 12 for RISC-V, so having a hard cut-off for the compiler version isn't much of a problem.  On the other, GCC 11 is "only" three years old, which for certain applications may still be a no-go.

Reply:
Yeah, I was probably a bit hasty on this suggestion. :laughing: . You're totally right, and especially if the Ubuntu 22.04 LTS toolchain is GCC 10 then it will be with us for a while. Better if it can be made to work.

(I was a bit confused about how the older gcc can compile interrupts.c without specifying ZICSR. For anyone else following at home, these extensions [were part of "I" but got spun out to a new extension](https://patchwork.kernel.org/project/linux-riscv/patch/20230809165648.21071-1-xingmingzheng@iscas.ac.cn/)).